File: backend/app/schemas/transaction.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class TransactionResponse(BaseModel):
    id: UUID

    listing_id: UUID
    offer_id: UUID | None

    buyer_id: UUID
    seller_id: UUID

    agreed_price: Decimal
    currency: str
    quantity: int

    status: str

    buyer_confirmed_at: datetime | None
    seller_confirmed_at: datetime | None
    completed_at: datetime | None
    cancelled_at: datetime | None

    created_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )


class TransactionCancelRequest(BaseModel):
    reason: str = Field(
        min_length=3,
        max_length=1000,
    )

File: frontend/src/types/transaction.ts
export type MarketplaceTransaction = {
  id: string;
  transaction_number: string;

  listing_id: string;
  offer_id: string | null;

  buyer_id: string;
  seller_id: string;

  agreed_price: string;
  currency: string;
  quantity: number;

  status: string;

  buyer_confirmed_at: string | null;
  seller_confirmed_at: string | null;
  completed_at: string | null;
  cancelled_at: string | null;

  created_at: string;
};